src/codegen_call_native.cpp

```
@@ -31,25 +31,27 @@ llvm::Value *CodeGen::emitTableDrivenNativeCall(
     // --- Special case: variadic arity (e.g. path::join) ---
     if (entry->arity == -1) {
         size_t n = e.args.size();
-        if (n < 2 || n > 4)
-            codegenError(std::string(entry->fn_name) + "() requires 2, 3, or 4 arguments");
 
-        // Match signature by arity (same validation as fixed-arity path)
+        // Check sig key BEFORE arity validation so that a name collision
+        // with a different-library function falls through instead of erroring.
         std::string sigKey = nativeSigKey(package, e.callee);
```

Clang (14.0.6)

[warning] variable 'sigKey' is not initialized

(cppcoreguidelines-init-variables)